flytekit/tools/translator.py

```diff
@@ -820,21 +819,73 @@ def get_serializable(
     else:
         raise ValueError(f"Non serializable type found {type(entity)} Entity {entity}")
 
-    if isinstance(entity, TaskSpec) or isinstance(entity, WorkflowSpec):
-        # 1. Check if the size of long description exceeds 16KB
-        # 2. Extract the repo URL from the git config, and assign it to the link of the source code of the description entity
-        if entity.docs and entity.docs.long_description:
-            if entity.docs.long_description.value:
-                if sys.getsizeof(entity.docs.long_description.value) > 16 * 1024 * 1024:
-                    raise ValueError(
-                        "Long Description of the flyte entity exceeds the 16KB size limit. Please specify the uri in the long description instead."
-                    )
-            entity.docs.source_code = SourceCode(link=settings.git_repo)
-    # This needs to be at the bottom not the top - i.e. dependent tasks get added before the workflow containing it
+    if (
+        isinstance(entity, (PythonTask, WorkflowBase))
+        and not isinstance(entity, ReferenceEntity)
+        and entity.module_file
+    ):
+        # Extract the repo URL from the git config, and assign it to the link of the source code of the description entity
+        entity.docs.source_code = SourceCode(link=_get_git_link(str(entity.module_file), settings))
```

Comment:
`module_file` is already a string

```suggestion
        entity.docs.source_code = SourceCode(link=_get_git_link(entity.module_file, settings))
```